### ISSUE TITLE
Go-karts, dodgems, flying saucers (+variants) use vehicle name/description in the build window

### ISSUE THREAD

**Issue by Wuzzy2** [OpenRCT2/OpenRCT2#10194]
**OS:** GNU/Linux
**Version:** 0.2.3
**Commit/Build:** stable

- [ ] Reproducible in RCT2 (vanilla)? No.
- [ ] Specific to multiplayer? No.

There are some rides which, in the build window, seem to use the same name+description as the *vehicle* name from the JSON. I think this is not a good idea because 1) it's weirdly inconsistent and 2) it forces OpenRCT2 to use the same string for both ride and vehicle which might not always make sense.

Here are the concrete problems in the build window:

* Go-Karts: The description *of the ride* is the same as for the default go-kart vehicle, which is just “Self-drive pretrol-engined go karts”. This describes the vehicle, not the ride. It would make sense to write a separate description for the go-karts *ride*
* Dodgems: Description *of the ride* is “Self-drive electric dodgem cars”. Same problem.
* Flying Saucers, Chinese Dragonhead, Flower Power, Skidoo Dodgems: All these descriptions are describing the vehicle, not the ride

In all these cases, one possible solution is to check if RCT2 (vanilla) actually includes description strings for the rides themselves (rather than the vehicle). If yes, then use those instead of the JSONs.

In simpler terms:
* In the build window for the rides above, use a description that is more appropriate for the build window (maybe by taking the RCT2 vanilla text)
* In the vehicles window, use the JSON text (already done, no change needed)

See also: https://github.com/OpenRCT2/objects/issues/50

**Comment by Gymnasiast** [comment]
You can fix the go karts by modifying the object description, since it is a tracked ride. (Like you already did for many other vehicles.)

The flat rides are a bit harder.

> In all these cases, one possible solution is to check if RCT2 (vanilla) actually includes description strings for the rides themselves (rather than the vehicle). If yes, then use those instead of the JSONs.

What do you mean? Do you mean using a generic description that applies to all "vehicles"/variants?

**Comment by Wuzzy2** [comment]
> You can fix the go karts by modifying the object description, since it is a tracked ride.

You mean `rct2/ride/rct2.kart1.json`, right?

It's not as simple as that. Currently, the string “Self-drive pretrol-engined go karts” is used *both* in the vehicle description selection and in the build window. This description is correctly describing the vehicle, but it's inappropriate in the ride window. This might be a bug.
If I would change this string (for example: “Riders race on a circular track”), the reverse is true: The string will be appropriate for the build window but not really for the vehicle window.

> What do you mean? Do you mean using a generic description that applies to all "vehicles"/variants?

No. Each vehicle still has its own description, that should still be used. I am mostly concerned what you see in the “build ride” window, when you hover a ride. Currently, the description in the build window for certain rides is identical to the vehicle description. This seems like a bug, because its not the case for most other rides, like roller coasters. For all roller coasters, the description in build window is independent from the description of the individual vehicle types.

Example:
Dodgems.

Vehicle description is “Self-drive electric dodgem cars”. But if you hover it in build window, you see the same description. But in build window it should describe the ride. Again, editing the JSON files will not fix this, as I would be changing *both* strings simultanously.

Counter-example:
Wooden roller coaster. If you hover this ride in vehicle window, you see a description that says what a wooden roller coaster is. If you then build a wooden roller coaster, then click on it and open the vehicle tab and select any vehicle, the descripition for these vehicles are actually describing the vehicle, not the roller coaster. This is how it currently works like.

For some reason, the go-karts, dodgems and flying saucers and variants are behaving oddly different. I do not see a logical reason for this kind of behaviour, therefore I claim a bug.

Long story short, I want the go-karts, dodgems and flying saucers and variants to have a *ride* description in the ride window rather than a vehicle description, just like every other attraction. The vehicle description of each of the vehicle would still be independent from each other.

**Comment by Gymnasiast** [comment]
> It's not as simple as that. Currently, the string “Self-drive pretrol-engined go karts” is used both in the vehicle description selection and in the build window.

No, it isn't. You can safely modify the string in the object and it will only apply to the vehicle tab. The string in the New Ride window is stored in `en-GB.txt`. It just so happens that both strings are currently set to the same value.

**Comment by Wuzzy2** [comment]
Ohhhhh! I didn't realize. Thanks for telling me that.

OK, time to collect suggestions for improved ride descriptions (will result in PR when I get an official OK):

* Go-Karts: “Asphalt track on which riders race each other in go-karts”

* Dodgems: “Glass building in which guests bump into each other in electric dodgem cars”
* Triceratops Dodgems: “Glass building in which guests horns with each other other in triceratops dodgem cars”
* Fighting Knights Dodgems: “Glass building in which riders joust in horse-shaped dodgem cars”

* Flying Saucers: “Platform on which guests freely ride little saucer-shaped hovercraft vehicles”
* Flower Power Ride: “Platform on which guests bump into each other in flower-shaped hovercraft vehicles”
* Cyclops Ride: “Platform on which guests freely ride the Eye of a Giant Cyclops”
* Chinese Dragonhead Ride: “Platform on which guests battle each other in dragonhead-shaped cars”
* Skidoo Dodgems: “Platform on which guests freely ride in small skidoo-shaped vehicles”

But this makes me wonder: While I can find the strings for dodgems, go-karts and flying saucers in `en-GB.txt`, I don't find anything for the “variant rides” (flower power, etc.). So maybe I was partially right.

**Comment by Wuzzy2** [comment]
Also note that attractions with the word “Ride” in them seem to take their name from the JSONs rather than `en-GB.txt`.

For example, “Cyclops Ride”. This name affects both the *ride* name as well the *vehicle* name. Ideally I'd prefer to call the *ride* “Cyclops Ride” (as before) but the vehicle something like “Cyclops Car” or “Cyclops Vehicle” or maybe even just “Cyclops” or something like that.

Similar rationale for Flower Power and Chinese Dragonhead.

**Comment by Gymnasiast** [comment]
> Also note that attractions with the word “Ride” in them seem to take their name from the JSONs rather than en-GB.txt.

That is correct, yes. Flat rides, like the merry-go-round, dodgems and all shops, are listed independently in the New Ride menu, while tracked rides are not. Flat rides also take their names and descriptions in the New Ride window from the JSON file, while tracked rides take them from `en-GB.txt`.

While your solution of adding separate vehicle tab and New Ride window names and descriptions would certainly _work_, I'm not yet sure if it is the best way to solve it, as it requires a lot of extra, very similar text that every translator would have to translate. I think I have an alternative, but I need to work it out a bit. I'll come back to this.

**Comment by Wuzzy2** [comment]
Another possibility would be to use names that would work both as ride and vehicle names. But I'm not sure if this is even possible. Maybe it's a bit easier for the descriptions.

Note that I made the proposal only for those 9 specific rides/vehicles I listed above. I think other rides are not affected by the problem.

**Comment by Wuzzy2** [comment]
OK, I just tried editing `en-GB.txt`. Editing `STR_0534` indeed affected the go-kart ride description, as you said.

However, I also tried editing `STR_0537` and `STR_0582` to edit the description of Dodgems and Flying Saucers (the default rides). But this didn't change anything. Apparently these 2 strings are ignored. \*shrug\*

**Comment by Wuzzy2** [comment]
The go-kart problem is solved by editing `en-GB.txt` accordingly.
The remaining problem is dodgems and flying saucers (+variants)

One solution is to simply rewrite the descriptions for dodgems and flying saucers (+variants) so that they make sense both in ride window as well in vehicle window. This is probably the most lightweight change as no code needs to be touched (only JSONs).

My attempt at generic descriptions for dodgems and flying saucers (+variants):

* Dodgems: “Guests bump into each other in self-drive electric dodgem cars”
* Triceratops Dodgems: (no change)
* Fighting Knights Dodgems: (no change)
* Flying Saucers: “Guests ride in little saucer-shaped hovercraft vehicles that they freely control”
* Flower Power Ride: “Guests ride in flower-shaped hovercraft vehicles that they freely control”
* Cyclops Ride: (no change)
* Chinese Dragonhead Ride: “Guests battle each other in dragonhead-shaped cars”
* Skidoo Dodgems: “Guests slide around in small skidoo-shaped vehicles that they freely control”

Upside: These strings are written so they would (kinda) make sense both as ride as well as vehicle description.
Downside: The ride-vs-vehicle name conflict still remains (e.g. “Chinese Dragonhead Ride” for both ride and vehicle).

**Comment by Wuzzy2** [comment]
@Gymnasiast: Damn, I just noticed this issue has NOT been fixed by “Simplify most vehicle descriptions” (objects#55). But to be fair, this PR did not make it work (at least), but it didn't really improve it either. So this issue can still not be closed.

In this PR you requested to revert to the vehicle-style descriptions for the dodgems and flying saucers (+variants), because you claimed it was no longer neccessary, and I complied. Today I have actually made tests and noticed that this is still not the case, so my revert was pointless …

The dodgems and flying saucers rides (+variants) still use the vehicle description both for the vehicles and in the ride window. This is a little bit sub-obtimal because it just doesn't describe the ride well.

If you do not believe me, do this to test my claims:

1. Open the cheat menu, disable research limits
2. Open the cheaty object selection
3. Hover all the dodgems and flying saucer vehicles and read their descriptions
4. Activate all dodgems and flying saucer vehicles
5. Now open the build ride window and hover all dodgems and flying saucer-style rides and read their descriptions
6. Compare what you read in step 3 and 5
7. Do you think the description in the ride window is nice?

I still think many ride descriptions (for dodgems/flying saucer/variants) aren't that nice. I still think my previously suggested idea of writing the vehicle descriptions of dodgems and flying saucer *vehicles* in a generalized style is the best one.

How good are my chances if I would post a SMALL follow-up PR that only touches this thing?

**Comment by Wuzzy2** [comment]
Correction: In my previous post I said:

> But to be fair, this PR did not make it work (at least)

I meant to say:

> But to be fair, this PR did not make it *worse* (at least)

**Comment by AaronVanGeffen** [comment]
(For what it's worth, you can edit posts with the [...] dropdown button in the top-right corner of a message.)

**Comment by Gymnasiast** [comment]
> In this PR you requested to revert to the vehicle-style descriptions for the dodgems and flying saucers (+variants), because you claimed it was no longer neccessary, and I complied. Today I have actually made tests and noticed that this is still not the case, so my revert was pointless …

Except that I didn't claim that at all. Let's quote what I said in https://github.com/OpenRCT2/objects/pull/55#issuecomment-605655507 :

> There is no need for that. The description shown for the ride is stored in en-GB.txt (except for **flat rides** and shops).

Dodgems and flying saucers are flat rides.

**Comment by Wuzzy2** [comment]
Oops, I have misunderstand you. But the remainder of my post is still true. Also note that currently, ALL vehicle variants of dogems and flying saucers will show as separate buildings in the build window (probably intentional).

Let's look how the descriptions of the BUILD window currently look like:

* Dodgems: Self-drive electric dodgems
* Fighting Knights Dodgems: Riders joust on horse-themed dodgems
* Triceratops Dodgems: Riders lock horns with each other in triceratops dodgems
* Chinese Dragonhead Ride: Self-drive electric dodgems
* Flower Power Ride: Riders ride in flower-shaped hovercraft vehicles that they freely control
* Skidoo Dodgems: Self-drive skidoo-shaped dodgems
* Cyclops Ride: Riders ride the Eye of a Giant Cyclops
* Flying Saucers: Self-drive saucer-shaped hovercraft vehicles

Note the inconsistent writing style. All these descriptions work fine as vehicle descriptions, but some of them are a bit weird in the Build Ride window.
The descrition of Fighting Knights Dodgems and Triceratops Dodgems is really nice, as it works for both vehicle and the ride window. But e.g. the Dodgems description sounds weird and oddly generic when read in the Build window.

**Basically this whole issue is about making the writing style (1) consistent and (2) functional both as vehicle and for the build window.**

Therefore, I repeat my suggestion of rewriting these strings to what I have written a few posts ago.

Also, sorry for dragging this issue on for sooooo long, but I'm a little bit perfectionistic here. ;-)

**Comment by Gymnasiast** [comment]
I don't mind your persistence or perfectionism. I do have to say that you sometimes come across as a bit aggressive or agitated (especially rhetorical questions like "Do you think the description in the ride window is nice?"). I'm sure that's not how you intend it, but it is something to watch out for, since it makes discussing this not quite as pleasant as it could be.

Back the issue at hand. You can open another pull request in OpenRCT2/objects to change the descriptions. Personally, I have to say, I don't really mind if a description is a bit too geared towards the New Ride window or the vehicle tab, but I welcome any PR to improve this.

Of course, this is assuming that we keep the technical side as-is. Perhaps there is a case to be made for grouping the different dodgems and flying saucers together, rendering this issue moot?
